packages/react-bootstrap-table2/src/body.js

```diff
@@ -35,7 +35,10 @@ const Body = (props) => {
 
   if (isEmpty) {
     const indication = _.isFunction(noDataIndication) ? noDataIndication() : noDataIndication;
-    content = <RowSection content={ indication } colSpan={ visibleColumnSize } />;
+    const colSpan = visibleColumnSize + (
+      selectRow.mode !== Const.ROW_SELECT_DISABLED && !selectRow.hideSelectColumn ? 1 : 0
+    );
+    content = <RowSection content={ indication } colSpan={ colSpan } />;
```

Comment:
Hi @patrickomeara I think the best solution is following:

in the `packages/react-bootstrap-table2/src/props-resolver.js` and change as below
```js
class ColumnResolver extends ExtendBase {
    visibleColumnSize(onlyCheckColumns = false) {
      let columnLen = this.props.columns.filter(c => !c.hidden).length;
      if (onlyCheckColumns) return columnLen;
      if (this.props.selectRow && !this.props.selectRow.hideSelectColumn) {
        columnLen += 1;
      }
      return columnLen;
    }
  };
```

so that I think we don't change the `body.js`, handle `visibleColumnSize` well can solve any column length issue.

However, I also saw there's a something you can help to fix:
in the `packages/react-bootstrap-table2/src/index.js`, Line:12 and change as below
```js
validateProps() {
      const { keyField } = this.props;
      if (!keyField) {
        throw new Error('Please specify a field as key via keyField');
      }
      if (this.visibleColumnSize(true) <= 0) {
        throw new Error('No any visible columns detect');
      }
    }
```